tests/lib/nested.sh

```diff
@@ -619,10 +619,19 @@ defaults:
     journal:
       persistent: true
 EOF
+            local GADGET_EXTRA_CMDLINE=""
             if [ "$NESTED_SNAPD_DEBUG_TO_SERIAL" = "true" ]; then
                 # add snapd debug and log to serial console for extra
                 # visibility what happens when a machine fails to boot
-                echo 'console=ttyS0 snapd.debug=1 systemd.journald.forward_to_console=1 ' > pc-gadget/cmdline.extra
+                GADGET_EXTRA_CMDLINE="snapd.debug=1 systemd.journald.forward_to_console=1"
+            fi
+            if [ -n "$NESTED_EXTRA_CMDLINE" ]; then
+                GADGET_EXTRA_CMDLINE="$GADGET_EXTRA_CMDLINE $NESTED_EXTRA_CMDLINE"
+            fi
+
+            if [ -n "$GADGET_EXTRA_CMDLINE" ]; then
+                echo "Configuring command line parameters in the gadget snap: \"console=ttyS0 $GADGET_EXTRA_CMDLINE\""
+                echo "console=ttyS0 $GADGET_EXTRA_CMDLINE" > pc-gadget/cmdline.extra
```

Comment:
Valentin already pointed this out, but it seems to me the logic here has changed, the `console=ttyS0` was moved  from the DEBUG_TO_SERIAL to here where it's now only included if GADGET_EXTRA_CMDLINE is not set, was this intentional?

Reply:
right, I'll update that, thanks for the review